File: C-Sharp/3.cs
// Longest substring without repeating characters.

public class Solution {
    public int LengthOfLongestSubstring(string s)
    {
        int maxValue = 0;
        for (int i = 0; i < s.Length; i++)
        {
            int counter = 0;
            HashSet<char> hashSet = new HashSet<char>();
            for (int j = i; j < s.Length; j++)
            {
                if (!hashSet.Add(s[j])) break;

                counter++;

                if (counter >= maxValue) maxValue = counter;
            }
        }
        return maxValue;
    }
}

File: Python/3.py
# Longest substring without repeating characters.

class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        max_value = 0
        for i in range(len(s)):
            counter = 0
            hash_set = set()
            for j in range(i, len(s)):
                if s[j] in hash_set:
                    break
                hash_set.add(s[j])
                counter += 1
                if counter >= max_value:
                    max_value = counter
        return max_value
